lib/erb_lint/linters/hard_coded_string.rb

```diff
@@ -60,14 +68,53 @@ def autocorrect(processed_source, offense)
         return unless string.strip.length > 1
 
         corrector = klass.new(processed_source.filename, offense.source_range)
-        node = RuboCop::AST::StrNode.new(:str, [string])
+
+        node = RuboCop::ProcessedSource.new(
+          replace_erb_symbols_for_interpolation(string),
+          RUBY_VERSION.to_f,
+        ).ast
+
+        node = node.each_child_node(:str).first if node.type != :str
```

Comment:
Pick the first `str` child_node since `shopify-cops` assumes the error is on a `str` node (and not on `dstr`)